### PR TITLE
Add dependencies for top-level build script

### PR DESCRIPTION
**Pull request by alexcrichton**
Helps avoid unnecessarily compiling the main library!

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -8,6 +8,8 @@ use dotenv::dotenv;
 use std::env;
 
 fn main() {
+    println!("cargo:rerun-if-env-changed=TET_DATABASE_URL");
```
Comment (kureuil) [inline build.rs]: Isn't this supposed to be `TEST_DATABASE_URL` ? :)

Comment (alexcrichton) [inline build.rs]: Oops indeed!

**Comment by carols10cents** [comment]
awesome!!! thanks!! 

bors: r+

**Comment by jtgeibel** [comment]
I've just tested this locally, by: running `cargo test`, adding a migration, and running `cargo test` again.  My new migration was not run.

I've added the following to the build script, which works for my simple test, but I'm not sure it is a complete solution:

```rust
println!("cargo:rerun-if-changed=migrations/");
```

Here are the reasons I think this may not be complete:

* The build script documentation notes that the behavior of timestamps on directories can be platform specific.  However, I'm not aware of what those differences may be.
* The documentation notes that the directory is not traversed to check for additional timestamp changes.  We can manually walk the tree and add a line for each directory and file, but I'm not certain that is necessary for this use case.  If someone changes a migration during development (after running the initial migration), then they will need to run `diesel migration redo` before running `cargo test` anyway.

Thoughts?

**Comment by carols10cents** [comment]
@jtgeibel I would like a pull request for that please, I think it'll be good enough for most of the cases!

**Pull request merged**